interface/patient_file/summary/add_edit_issue.php

```diff
@@ -230,86 +231,49 @@ function ActiveIssueCodeRecycleFn($thispid2, $ISSUE_TYPES2)
+    $form_return = !empty($_POST['form_return']) ? DateToYYYYMMDD($_POST['form_return']) : null;
 
     $form_injury_part = $_POST['form_medical_system'] ?? '';
     $form_injury_type = $_POST['form_medical_type'] ?? '';
 
-    if ($issue) {
-        $query = "UPDATE lists SET " .
-            "type = '"        . add_escape_custom($text_type)                  . "', " .
-            "title = '"       . add_escape_custom($_POST['form_title'])        . "', " .
-            "udi = '"         . add_escape_custom($_POST['form_udi'])               . "', " .
-            "udi_data = '"    . add_escape_custom($_POST['udi_data'])          . "', " .
-            "comments = '"    . add_escape_custom($_POST['form_comments'])     . "', " .
-            "begdate = "      . QuotedOrNull($form_begin)   . ", "  .
-            "enddate = "      . QuotedOrNull($form_end)     . ", "  .
-            "returndate = "   . QuotedOrNull($form_return ?? null)  . ", "  .
-            "diagnosis = '"   . add_escape_custom($_POST['form_diagnosis'])    . "', " .
-            "occurrence = '"  . add_escape_custom($_POST['form_occur'])        . "', " .
-            "classification = '" . add_escape_custom($_POST['form_classification']) . "', " .
-            "reinjury_id = '" . add_escape_custom($_POST['form_reinjury_id'] ?? '')  . "', " .
-            "referredby = '"  . add_escape_custom($_POST['form_referredby'])   . "', " .
-            "injury_grade = '" . add_escape_custom($_POST['form_injury_grade'] ?? '') . "', " .
-            "injury_part = '" . add_escape_custom($form_injury_part)           . "', " .
-            "injury_type = '" . add_escape_custom($form_injury_type)           . "', " .
-            "outcome = '"     . add_escape_custom($_POST['form_outcome'])      . "', " .
-            "destination = '" . add_escape_custom($_POST['form_destination'])   . "', " .
-            "reaction ='"     . add_escape_custom($_POST['form_reaction'])     . "', " .
-            "verification ='"     . add_escape_custom($_POST['form_verification'])     . "', " .
-            "severity_al ='"     . add_escape_custom($_POST['form_severity_id'])     . "', " .
-            "list_option_id ='"     . add_escape_custom($_POST['form_title_id'])     . "', " .
-            "erx_uploaded = '0', " .
-            "modifydate = NOW() " .
-            "WHERE id = '" . add_escape_custom($issue) . "'";
-        sqlStatement($query);
-        if ($text_type == "medication" && $form_end != '') {
-            sqlStatement(
-                'UPDATE prescriptions SET '
-                . 'medication = 0 where patient_id = ? '
-                . " and upper(trim(drug)) = ? "
-                . ' and medication = 1',
-                array($thispid, strtoupper($_POST['form_title']))
-            );
+    $issueRecord = [
+        'type' => $text_type
+        // I don't like how these date columns use this quasi 'NULL' value but its how the underlying service works...
```

Comment:
The original sql queries are using null without quotes though (they are run the the QuotedOrNull() function to ensure that the null does not get the quotes).

Reply:
oh, that QuotedOrNull() function is weird.
It doesn't seem right to do that. I wonder if that got wrongly updated during sql-injection work in past.

Reply:
nope, I was wrong. That return "NULL" code has been there since the beginning of time.

Reply:
Just want to make sure from your comment, this is ok to do it the way I'm doing here right?  The values still get set to NULL when they are saved.

Reply:
yep, looks good